eth/tracers/native/call_flat.go

```diff
@@ -0,0 +1,346 @@
+	tracer tracers.Tracer
+	// env    *vm.EVM
+	config flatCallTracerConfig
+	ctx    *tracers.Context // Holds tracer context data
+	// callstack         []callParityFrame
+	// interrupt         uint32           // Atomic flag to signal execution interruption
+	reason error // Textual reason for the interruption
+	// activePrecompiles []common.Address // Updated on CaptureStart based on given rules
+}
+
+type flatCallTracerConfig struct {
+	ConvertedParityErrors bool `json:"convertedParityErrors"` // If true, call tracer converts errors to parity format
+}
+
+// newFlatCallTracer returns a new mux tracer.
+func newFlatCallTracer(ctx *tracers.Context, cfg json.RawMessage) (tracers.Tracer, error) {
+	var config flatCallTracerConfig
+	if cfg != nil {
+		if err := json.Unmarshal(cfg, &config); err != nil {
+			return nil, err
+		}
+	}
+
+	tracer, err := tracers.New("callTracer", ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &flatCallTracer{tracer: tracer, ctx: ctx, config: config}, nil
+}
+
+// CaptureStart implements the EVMLogger interface to initialize the tracing operation.
+func (t *flatCallTracer) CaptureStart(env *vm.EVM, from common.Address, to common.Address, create bool, input []byte, gas uint64, value *big.Int) {
+	t.tracer.CaptureStart(env, from, to, create, input, gas, value)
+}
+
+// CaptureEnd is called after the call finishes to finalize the tracing.
+func (t *flatCallTracer) CaptureEnd(output []byte, gasUsed uint64, elapsed time.Duration, err error) {
+	t.tracer.CaptureEnd(output, gasUsed, elapsed, err)
+}
+
+// CaptureState implements the EVMLogger interface to trace a single step of VM execution.
+func (t *flatCallTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, rData []byte, depth int, err error) {
+	t.tracer.CaptureState(pc, op, gas, cost, scope, rData, depth, err)
+}
+
+// CaptureFault implements the EVMLogger interface to trace an execution fault.
+func (t *flatCallTracer) CaptureFault(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, depth int, err error) {
+	t.tracer.CaptureFault(pc, op, gas, cost, scope, depth, err)
+}
+
+// CaptureEnter is called when EVM enters a new scope (via call, create or selfdestruct).
+func (t *flatCallTracer) CaptureEnter(typ vm.OpCode, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int) {
+	t.tracer.CaptureEnter(typ, from, to, input, gas, value)
+}
+
+// CaptureExit is called when EVM exits a scope, even if the scope didn't
+// execute any code.
+func (t *flatCallTracer) CaptureExit(output []byte, gasUsed uint64, err error) {
+	t.tracer.CaptureExit(output, gasUsed, err)
+}
+
+func (t *flatCallTracer) CaptureTxStart(gasLimit uint64) {
+	t.tracer.CaptureTxStart(gasLimit)
+}
+
+func (t *flatCallTracer) CaptureTxEnd(restGas uint64) {
+	t.tracer.CaptureTxEnd(restGas)
+}
+
+// GetResult returns an empty json object.
+func (t *flatCallTracer) GetResult() (json.RawMessage, error) {
```

Comment:
Here we're doing an extra marshal and unmarshal. A call to `t.tracer.GetResult` is not necessary. We can directly access its callstack and process the output.

Reply:
Yes, this make things a lot easier. Will try to work on this direction using this proxy tracer and see how it goes. 
Will be nice if you can reply here and with a go/no-go.